### PR TITLE
Docs(config) : add module.unsafeCache to v4 docs

### PR DESCRIPTION
**Pull request by salemhilal**
This PR mirrors #3714 for the v4 documentation.
It addresses the following issues:
- #2539
- webpack/webpack#10771

### CODE REVIEW

File: src/content/configuration/module.mdx [hunk 1]
```diff
@@ -53,6 +55,28 @@ module.exports = {
...
+```javascript
+module.exports = {
+  //...
+  module: {
+    unsafeCache: false,
+  }
+};
+```
+
+W> In Webpack 5, the default value is `true` if the `cache` option is enabled _and_ the module appears to come from the node_modules directory.
```
Comment (montogeek) [inline src/content/configuration/module.mdx:75]:
```suggestion
W> In webpack 5, the default value is `true` if the `cache` option is enabled _and_ the module appears to come from the node_modules directory.
```

Although this documentation is for webpack 5 by default, so the message should be for v4 instead.

Comment (salemhilal) [reply src/content/configuration/module.mdx:75]: You had me worried!

File: src/content/configuration/module.mdx [hunk 2]
```diff
@@ -53,6 +55,28 @@ module.exports = {
...
+  module: {
+    unsafeCache: false,
+  }
+};
+```
+
+W> In webpack 5, the default value is `true` if the `cache` option is enabled _and_ the module appears to come from the node_modules directory.
+
+
+
```
Comment (EugeneHlushko) [inline src/content/configuration/module.mdx:78]: need to clean up unnecessary rows

Comment (salemhilal) [reply src/content/configuration/module.mdx]: Done

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**